src/extent.c

```diff
@@ -312,11 +316,17 @@ extents_insert_locked(tsdn_t *tsdn, extents_t *extents, extent_t *extent,
 	size_t size = extent_size_get(extent);
 	size_t psz = extent_size_quantize_floor(size);
 	pszind_t pind = sz_psz2ind(psz);
-	if (extent_heap_empty(&extents->heaps[pind])) {
-		bitmap_unset(extents->bitmap, &extents_bitmap_info,
-		    (size_t)pind);
+	if (config_sized_alloc_region && extent_in_sized_alloc_region(extent)) {
+		szind_t szind = extent_szind_get(extent);
+		extent_heap_insert(&extents->sized_alloc_region_heaps[szind],
+		    extent);
+	} else {
+		if (extent_heap_empty(&extents->other_heaps[pind])) {
+			bitmap_unset(extents->bitmap, &extents_bitmap_info,
```

Comment:
Oh, that's tricky. I think I'm OK just declaring that the sized-alloc-region is reserved for auto arenas, letting us sidestep the issue.

If this turns out to be a problem for some reason, we can always add a fallback global extents_t.